lib/xboxrt/vcruntime/dynamic_cast.cpp

```diff
@@ -0,0 +1,55 @@
+#include <windows.h>
+#include <excpt_cxx.hpp>
+
+struct RTTIBaseClassDescriptor
+{
+    TypeDescriptor *pTypeDescriptor;
+    DWORD numContainedBases;
+    PMD where;
+    DWORD attributes;
+};
+
+struct RTTIBaseClassArray
+{
+    RTTIBaseClassDescriptor *arrayOfBaseClassDescriptors[];
+};
+
+struct RTTIClassHierarchyDescriptor
+{
+    DWORD signature;
+    DWORD attributes;
+    DWORD numBaseClasses;
+    RTTIBaseClassArray *pBaseClassArray;
+};
+
+struct RTTICompleteObjectLocator
+{
+    DWORD signature;
+    DWORD offset;
+    DWORD cdOffset;
+    TypeDescriptor *pTypeDescriptor;
+    RTTIClassHierarchyDescriptor *pClassDescriptor;
+};
+
+static const DWORD *get_vftable (void *objptr)
+{
+    return *static_cast<const DWORD **>(objptr);
+}
+
+static const RTTICompleteObjectLocator *getobjlocator (void *objptr)
+{
+    const DWORD *vftable = get_vftable(objptr);
+    return reinterpret_cast<const RTTICompleteObjectLocator *>(vftable[-1]);
+}
+
+extern "C" __cdecl void *__RTCastToVoid (void *objptr)
+{
+    if (!objptr) return nullptr;
```

Comment:
I'd prefer `{}` and a linebreak, to avoid a goto-fail. I happened to come another goto-fail bug this week (while reviewing code in an unrelated private project; what I'm saying is: dangerous).